src/coreclr/jit/fgbasic.cpp

```diff
@@ -988,6 +988,20 @@ void Compiler::fgFindJumpTargets(const BYTE* codeAddr, IL_OFFSET codeSize, Fixed
 
                     if ((methodFlags & CORINFO_FLG_JIT_INTRINSIC) != 0)
                     {
+                        if (intrinsicID == CORINFO_INTRINSIC_StringLength)
+                        {
+                            if (makeInlineObservations && pushedStack.IsStackAtLeastOneDeep())
```

Comment:
We shouldn't need the `makeInlineObservations` check here, we do the following up on L956 (https://github.com/dotnet/runtime/pull/50784/files#diff-3984c3b964e9d53aaef46d61d7922eea89c4e61323f3a6dc563cbe2b851b3471L956):
```
if (!makeInlineObservations)
{
    break;
}
```